packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -159,17 +161,41 @@ function isEditableColumn(column: FieldSchema) {
 
 export class ExternalRequest<T extends Operation> {
   private readonly operation: T
-  private readonly tableId: string
-  private datasource?: Datasource
-  private tables: { [key: string]: Table } = {}
+  private readonly source: Table | ViewV2
+  private datasource: Datasource
```

Comment:
Changing how instances of this class are constructor by using a static initialiser meant we could remove the ? from here and tidy up some code throughout the class. I generally like to make sure classes, once instantiated, are in a known-good state.

Reply:
This is nice - I like it